specsim/quick.py

```diff
@@ -435,6 +442,14 @@ def simulate(self,sourceType,sourceSpectrum,airmass=1.,nread=1.,expTime=None,dow
             signalMask = (results.nobj)[:,j] > 0
             (results.snr)[:,j] = np.zeros((ndown,))
             (results.snr)[signalMask,j] = (results.nobj)[signalMask,j]/np.sqrt(variance[signalMask])
+            # Compute calib in downsampled wave grid, it's a sum because
+            # nphot is a sum over orginal wave bins.
+            calib_downsampled = np.sum(camera.sourceCalib[:last].reshape(downShape),axis=1)
+            # Add inverse variance for camera
+            (results.camivar)[:,j] = calib_downsampled**2/(variance+(variance==0))
```

Comment:
Writing `x / (y + (y==0))` is a neat way to avoid tedious masked expressions, but it does assume that `x[y==0] == 0` without explicitly asserting this, so is a potential source of subtle bugs as the code that calculates `x` evolves.  I would be in favor of either adding an assertion to make the assumption explicit, e.g.
```
assert x[y==0], `Expected x == 0 when y == 0.`
```
or else explicitly assigning zero where `y==0`, e.g.
```
z = np.zeros_like(x)
mask = y != 0
z[mask] = x[mask] / y[mask]
```
On an aesthetic note, `y + (y==0)` is not dimensionally valid, and I guess this would actually raise an error if `y` had astropy units.